Source/Core/Core/HW/EXI/EXI_DeviceIPL.cpp

```diff
@@ -347,68 +290,88 @@ void CEXIIPL::TransferByte(u8& _uByte)
+    if (IN_RANGE(ROM))
+    {
+      if (!m_command.is_write())
+      {
+        u32 dev_addr = DEV_ADDR_CURSOR(ROM);
+        // Technically we should descramble here iff descrambling logic is enabled.
+        // At the moment, we pre-decrypt the whole thing and
+        // ignore the "enabled" bit - see CEXIIPL::CEXIIPL
+        data = m_rom[dev_addr];
 
-          if ((position >= 0x001AFF00) && (position <= 0x001FF474) && !m_FontsLoaded)
+        if ((dev_addr >= 0x001AFF00) && (dev_addr <= 0x001FF474) && !m_fonts_loaded)
+        {
+          if (dev_addr >= 0x001FCF00)
           {
-            if (position >= 0x001FCF00)
-            {
-              PanicAlertT("Error: Trying to access Windows-1252 fonts but they are not loaded. "
-                          "Games may not show fonts correctly, or crash.");
-            }
-            else
-            {
-              PanicAlertT("Error: Trying to access Shift JIS fonts but they are not loaded. "
-                          "Games may not show fonts correctly, or crash.");
-            }
-            m_FontsLoaded = true;  // Don't be a nag :p
+            PanicAlertT("Error: Trying to access Windows-1252 fonts but they are not loaded. "
+                        "Games may not show fonts correctly, or crash.");
           }
+          else
+          {
+            PanicAlertT("Error: Trying to access Shift JIS fonts but they are not loaded. "
+                        "Games may not show fonts correctly, or crash.");
+          }
+          // Don't be a nag
+          m_fonts_loaded = true;
         }
       }
+    }
+    else if (IN_RANGE(SRAM))
+    {
+      u32 dev_addr = DEV_ADDR_CURSOR(SRAM);
+      if (m_command.is_write())
+        g_SRAM.raw[dev_addr] = data;
       else
+        data = g_SRAM.raw[dev_addr];
+    }
+    else if (IN_RANGE(UART))
+    {
+      switch (DEV_ADDR(UART))
+      {
+      case 0:
+        // Seems to be 16byte fifo
+        UartFifoAccess(&data);
+        break;
+      case 0xc:
+        // Seen being written to after reading 4 bytes from barnacle
+        break;
+      case 0x4c:
+        DEBUG_LOG(OSREPORT, "UART Barnacle %x", data);
+        break;
+      }
+    }
+    else if (IN_RANGE(WII_RTC))
+    {
+      // Wii only RTC flags... afaik only the Wii Menu initializes it
+      // Seems to be 4bytes at dev_addr 0x20
+    }
+    else if (IN_RANGE(EUART))
+    {
+      switch (DEV_ADDR(EUART))
       {
-        NOTICE_LOG(OSREPORT, "EXI IPL-DEV: %s %x at %08x", IsWriteCommand() ? "write" : "read",
-                   _uByte, m_uAddress);
+      case 0:
+        // Writes 0xf2 then 0xf3 on EUART init. Just need to return non-zero
+        // so we can leave the byte untouched.
+        break;
+      case 4:
+        UartFifoAccess(&data);
+        break;
       }
-      break;
+    }
+    else
+    {
+      NOTICE_LOG(EXPANSIONINTERFACE, "IPL-DEV Accessing unknown device");
     }
 
-    m_uRWOffset++;
+#undef DEV_ADDR_CURSOR
```

Comment:
Not sure about the use of defines instead of functions, here. Is there a noticeable speed up?

Reply:
I don't see a way to replace this with non-macro.